internal/stores/objects.go

```diff
@@ -523,10 +539,45 @@ func deleteObject(tx *gorm.DB, key string) error {
 func (s *SQLStore) get(key string) (dbObject, error) {
 	var obj dbObject
 	tx := s.db.Where(&dbObject{ObjectID: key}).
-		Preload("Slabs.Slab.Shards.Contracts.Host").
+		Preload("Slabs.Slab.Shards.DBSector.Contracts.Host").
 		Take(&obj)
 	if errors.Is(tx.Error, gorm.ErrRecordNotFound) {
 		return dbObject{}, ErrObjectNotFound
 	}
 	return obj, nil
 }
+
+// TODO: Should we return slabs which are below MinShards? I think so because
+// the caller should handle that and then manually delete them if necessary.
+func (s *SQLStore) slabsForRepair(n int, failureCutoff time.Time) ([]uint, error) {
+	failureQuery := s.db.Model(&dbSlab{}).
+		Select("id as slab_id").
+		Where("last_failure < ?", failureCutoff.UTC())
+	inner := s.db.Table("(?)", failureQuery).
+		Select("slab_id, db_sector_id as sector_id").
+		Joins("LEFT JOIN shards ON slab_id = shards.db_slab_id")
```

Comment:
should be `NATURAL JOIN` I think

Reply:
`NATURAL` joins default to an `INNER` join, which I think you want here because if not you might be left with a `1, NULL`  row here if there's no sectors for that slab for whatever reason, right? I don't think there's any harm in making this an `INNER` join it's just more correct, I think the same goes for the other `LEFT JOIN` but would have to check - I'll run it manually tomorrow

Reply:
Wouldn't an `INNER` join between the slabs and shards still return those `1, NULL` rows? 
Either way we should never end up with a shard that has a `NULL` sector id. Because you can't delete a sector without violating the foreign key constraint. We don't have a `OnDelete:SET NULL` defined on the sector.